app/controllers/games_controller.rb

```diff
@@ -50,4 +68,22 @@ def game_params
     params.require(:game).permit(:name, :white_player_id)
   end
 
+  def authenticate_turn
+    unless white_can_go? || black_can_go?
```

Comment:
@bhk1019 this would probably never trigger since the line means:
`if !(white_can_go? or black_can_go?)` where one of them would always be true.. so we'll need to find a way to show it if its not specific user's turn..

Reply:
This should work, since `white_can_go?` checks that both the user is logged in, the user is white, and it is white's turn. Similarly, `black_can_go?` checks that the user is logged in, the user is black, and it is black's turn. For example, if it's white's turn, but black is logged in, then since `white_player.id != current_user.id`, neither `white_can_go?` nor `black_can_go?` will return true, and the `unless` block will execute and block the action from running. Perhaps I should change the names of the functions to be more descriptive? Something like `white_is_logged_in_and_white_turn?` ?